File: TestPlay/TestPlay/ViewController.m
//
//  ViewController.m
//  TestPlay
//
//  Created by pandazheng on 15/9/5.
//  Copyright (c) 2015年 pandazheng. All rights reserved.
//

#import "ViewController.h"
#import <AVFoundation/AVFoundation.h>

@interface ViewController ()

@property (nonatomic,strong) AVPlayer* avPlayer;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    self.view.backgroundColor = [UIColor lightGrayColor];
    
    UIButton *playBtn = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    playBtn.frame = CGRectMake(40.0f, 40.0f, 200.0f, 48.0f);
    playBtn.center = self.view.center;
    playBtn.backgroundColor = [UIColor darkGrayColor];
    [playBtn setTitle:@"Play Video" forState:UIControlStateNormal];
    [playBtn setTitle:@"Play Video" forState:UIControlStateHighlighted];
    playBtn.titleLabel.font = [UIFont systemFontOfSize:32.0f];
    
    [playBtn addTarget:self action:@selector(onPlayButtonClicked:) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:playBtn];
}

- (NSUInteger) supportedInterfaceOrientations {
    return UIInterfaceOrientationMaskPortrait;
}

- (void) onPlayButtonClicked: (UIButton *) pSender {
    NSLog(@"button click......\n");
    NSString *videoPath = [[NSBundle mainBundle] pathForResource:@"Mac" ofType:@"mp4"];
    NSLog(@"%@",videoPath);
    self.avPlayer = [AVPlayer playerWithURL:[NSURL fileURLWithPath:videoPath]];
    
    AVPlayerLayer *playerLayer = [AVPlayerLayer playerLayerWithPlayer:self.avPlayer];
    self.avPlayer.actionAtItemEnd = AVPlayerActionAtItemEndNone;
    playerLayer.frame = self.view.bounds;
    //playerLayer.frame = CGRectMake(0, 0, self.view.frame.size.width, self.view.frame.size.height);
    [self.view.layer addSublayer:playerLayer];
    
    [self.avPlayer play];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: FmdbDemo/FmdbDemo/ViewController.swift
//
//  ViewController.swift
//  FmdbDemo
//
//  Created by pandazheng on 15/9/6.
//  Copyright (c) 2015年 pandazheng. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        test()
    }
    
    func test() {
        let documentsFolder = NSSearchPathForDirectoriesInDomains(NSSearchPathDirectory.DocumentDirectory, NSSearchPathDomainMask.UserDomainMask, true)[0] as! String
        let path = documentsFolder.stringByAppendingPathComponent("test.sqlite")
        
        println(path)
        
        let database = FMDatabase(path: path)
        
        if !database.open() {
            println("Unable to open database")
            return
        }
        
        if !database.executeUpdate("create table test(x text,y text,z text)", withArgumentsInArray: nil) {
            println("create table failed: \(database.lastErrorMessage())")
        }
        
        if !database.executeUpdate("insert into test(x,y,z) values(?,?,?)", withArgumentsInArray: ["a","b","c"]) {
            println("insert 1 table failed: \(database.lastErrorMessage())")
        }
        
        if !database.executeUpdate("insert into test(x,y,z) values(?,?,?)", withArgumentsInArray: ["e","f","g"]) {
            println("insert 2 table failed: \(database.lastErrorMessage())")
        }
        
        if let rs = database.executeQuery("select x,y,z from test", withArgumentsInArray: nil) {
            while rs.next() {
                let x = rs.stringForColumn("x")
                let y = rs.stringForColumn("y")
                let z = rs.stringForColumn("z")
                println("x = \(x);y = \(y);z = \(z)")
            }
        }
        else {
            println("select failed: \(database.lastErrorMessage())")
        }
        
        database.close()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

